lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+require_once dirname(__FILE__).'/res/OmiseApiResource.php';
+
+/**
+ * This class is not intended to be used directly from client code.
+ *
+ * @see OmiseCharge::search()
+ */
+class OmiseSearch extends OmiseApiResource
+{
+    const ENDPOINT = 'search';
+
+    private $dirty = true;
+    private $scope = null;
+    private $query = '';
+    private $conditions = array();
+    private $page = 1;
+    private $order = 'chronological';
+
+    public static function scope($scope)
+    {
+        return new OmiseSearch($scope);
+    }
+
+    protected function __construct($scope, $publickey = null, $secretkey = null)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->dirty = true;
+        $this->scope = $scope;
+    }
+
+    public function query($query)
+    {
+        $this->dirty = true;
+        $this->query = $query;
+        return $this;
+    }
```

Comment:
`mergeAttribute` with `$attributes` looks good. I also will move some `if` check to the setter instead. It may be even better if I `unset` empty fields.